### PR TITLE
Removes moment from reaction and uses luxon

### PR DESCRIPTION
**Pull request by kierangillen**
- This PR completely removes [moment](https://momentjs.com/) from Reaction and replaces it with [luxon](https://moment.github.io/luxon/)
- For context and reasons see: https://github.com/artsy/reaction/issues/2474

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 78aeb9d49797a9bce478aba3812fefe582cb1720

https://deploy-preview-2534--artsy-reaction.netlify.com

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit eca55f2cd40f634ebb2dfbca51235ee5427260ce

https://deploy-preview-2534--artsy-reaction.netlify.com

File: src/Apps/Order/Routes/__tests__/Accept.test.tsx [hunk 1]
```diff
@@ -74,12 +74,9 @@ describe("Accept seller offer", () => {
...
           order: {
             ...testOrder,
-            stateExpiresAt: moment(NOW)
-              .add(1, "day")
-              .add(4, "hours")
-              .add(22, "minutes")
-              .add(59, "seconds")
-              .toISOString(),
+            stateExpiresAt: DateTime.fromISO(NOW)
+              .plus({ days: 1, hours: 4, minutes: 22, seconds: 59 })
```
Comment (dblandin) [inline src/Apps/Order/Routes/__tests__/Accept.test.tsx]: Woah. This is cool :+1:

File: src/Components/Publishing/News/__tests__/NewsDateDivider.test.tsx [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
   beforeAll(() => {
-    Date.now = () => Date.parse("01 Jan 2009 00:00:00 EST")
+    Settings.now = () => new Date("01 Jan 2009 00:00:00 EST").valueOf()
+    Settings.defaultZoneName = "America/New_York"
   })
 
   afterAll(() => {
-    Date.now = dateNow
+    Settings.now = realNow
+    Settings.defaultZoneName = realDefaultZone
```
Comment (dblandin) [inline src/Components/Publishing/News/__tests__/NewsDateDivider.test.tsx]: This `Settings` object is awesome :+1:

**Comment by artsyit** [comment]
:rocket: PR was released in v16.17.0 :rocket:

**Pull request merged**